javaslang/src/main/java/javaslang/collection/Stack.java

```diff
@@ -26,14 +26,14 @@
  * <li>{@link #push(Object[])}</li>
  * <li>{@link #pushAll(Iterable)}</li>
  * </ul>
- *
+ * <p>
  * See Okasaki, Chris: <em>Purely Functional Data Structures</em> (p. 7 ff.). Cambridge, 2003.
  *
  * @param <T> component type
  * @author Daniel Dietrich
  * @since 2.0.0
  */
-public interface Stack<T> {
+public interface Stack<T> extends Traversable<T> {
```

Comment:
We should not extend Traversable. If we need the method, it has to be duplicated (without inheritance).

There was a long discussion some time ago whether we should remove the Stack interface or not. We decided to keep it as rudimentary standalone interface. It is not a collection but a contract for Stack-like collections. List is currently the only implementation.